week-1-ts-ecosystem/w01e03-object-oriented-structure/src/Products.ts

```diff
@@ -0,0 +1,81 @@
+export enum ProductType {
+  "BuyNow" = "Buy Now",
+  "Auction" = "Auction",
+  "Free" = "Free",
+}
+
+export abstract class Product {
+  #id: string;
+  #name: string;
+  #amount: number;
+  #type: ProductType;
+
+  constructor(id: string, name: string, type: ProductType, amount: number) {
+    this.#id = id;
+    this.#name = name;
+    this.#type = type;
+    this.#amount = amount;
+  }
+
+  get id() {
+    return this.#id;
+  }
+
+  get name() {
+    return this.#name;
+  }
+
+  get type() {
+    return this.#type;
+  }
+
+  get amount() {
```

Comment:
Jeśli traktujemy klasę `Product` jako taki typowy model a.k.a. `Record` to nie widzę zasadności ukrywania pól i tych getterów tutaj.
Nic w nich się nie dzieje.